gcovr/json_generator.py

```diff
@@ -31,21 +50,65 @@ def print_json_report(covdata, output_file, options):
 
     for no in sorted(covdata):
         gcovr_json_file = {}
-        gcovr_json_file['file'] = presentable_filename(covdata[no].filename, root_filter=options.root_filter)
+        gcovr_json_file['file'] = presentable_filename(covdata[no].filename,
+                                                       root_filter=options.root_filter)
         gcovr_json_file['lines'] = _json_from_lines(covdata[no].lines)
         gcovr_json_root['files'].append(gcovr_json_file)
 
-    write_json = json.dump
-    if options.prettyjson:
-        write_json = functools.partial(write_json, indent=PRETTY_JSON_INDENT, separators=(',', ': '), sort_keys=True)
-    else:
-        write_json = functools.partial(write_json, sort_keys=True)
+    _write_json_result(gcovr_json_root, output_file, options.prettyjson)
 
-    if output_file is not None:
-        with open(output_file, 'w') as output:
-            write_json(gcovr_json_root, output)
-    else:
-        write_json(gcovr_json_root, sys.stdout)
+
+#
+# Produce gcovr JSON summary report
+#
+def print_json_summary_report(covdata, output_file, options):
+
+    total_lines = 0
+    total_covered = 0
+
+    json_dict = {}
+    json_dict['current_working_directory'] = options.root
```

Comment:
Should the cwd be root, or root relative to the output file, or the actual current working directory? Should it be a relative or absolute path? Depends on the exact intention for this field.

Reply:
I think I placed it thinking that it will allow to differentiated multiple analysis from different trees; should we rename it to be `root` to be consistent?

Reply:
One of the problems we have with gcov is that it doesn't make quite clear which paths are relative to what, so I'd like to avoid making similar problems for downstream. I think that relevant designs could include:

* all file paths are relative to the summary report location, and no root/cwd is needed; or
* there is a root or cwd field that is relative to the summary report location, and the reported file paths are relative to the root – similar to the HTML `<base>` element.

  ```suggestion
      json_dict['root'] = os.path.relpath(options.root, output_file)
  ```

Both designs have things that are in their favour, but I think that most use cases are more interested in the path relative to root, not relative to the report.

In any case, it should be possible to correctly interpret the report without having to know how gcovr was invoked.